src/app/sign-in/sign-in.component.spec.ts

```diff
@@ -1,25 +1,95 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {
+  ComponentFixture,
+  fakeAsync,
+  TestBed,
+  tick,
+} from '@angular/core/testing';
+import { Observable, of, Subject } from 'rxjs';
+
+import { By } from '@angular/platform-browser';
+
+import { ReactiveFormsModule, FormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatInputModule } from '@angular/material/input';
+import { MatError } from '@angular/material/form-field';
+
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
 
 import { SignInComponent } from './sign-in.component';
 
+import { User } from '../user.model';
+
+import { AuthService } from '../auth.service';
+
+const authServiceMock = {
+  signIn: () => {},
+};
+
 describe('SignInComponent', () => {
   let component: SignInComponent;
   let fixture: ComponentFixture<SignInComponent>;
+  let authService: AuthService;
+  let router: Router;
+  let email;
+  let matErrorEl: HTMLElement;
 
-  beforeEach(async(() => {
+  beforeEach(() => {
     TestBed.configureTestingModule({
-      declarations: [ SignInComponent ]
-    })
-    .compileComponents();
-  }));
+      imports: [
+        ReactiveFormsModule,
```

Comment:
Actually you don't need all this modules, you need components declared inside them, so provide mocked components in declarations section